execution_workflows/CSI-UTR/bin/postprocess_differential.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+
+import sys
+import argparse
+import os
+from os import listdir
+from os.path import isfile, join
+import csv
+
+def parse_args(args=None):
+    Description = "Reformat CSI-UTR output file to the output file of differential challenge"
+    Epilog = "Example usage: python postprocess_differential.py <CSI_UTR_OUTPUT_FOLDER> <FILE_OUT>"
+
+    parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+    parser.add_argument("CSI_UTR_OUTPUT_FOLDER", help="CSI-UTR output folder that contains file for differential challenge.")
+    parser.add_argument("FILE_OUT", help="Name of output file for the differential challenge.")
+    return parser.parse_args(args)
+
+
+def convert_to_differential(csi_utr_output_folder, file_out):
+    """
+    This function reformats the txt CSI-UTR output file to files for differential challenge
+    :param csi_utr_output_folder: folder containing differential output file to be converted 
+    :param file_out: identification challenge output file
+    :return: N/A
+
+    example:
+    CSI     ENSGENE GENE_SYM        PSI1 (control)  PSI2 (srsf3)    deltaPSI (control-srsf3)        P-value FDR
+ENSMUSG00000015652:5736416_5736416-5736328      ENSMUSG00000015652      Steap1  1       1       0       1       1
+    """
+    files = [f for f in listdir(csi_utr_output_folder) if isfile(join(csi_utr_output_folder, f))]
+    # find the file in the format of condition1_VS_condition2_CSI.WITHINUTR.diff.txt"
+    for f in files:
+       if "diff.txt" in f:
+           output_file = join(csi_utr_output_folder, f)
+           break
+
+    differential_out = open(file_out, "wt")
+    
+    with open(output_file, 'r') as f:
+        reader = csv.reader(f, delimiter='\t')
+        # skip the header line
+        next(reader)
+        for line in reader:
+            gene = line[1]
+            pvalue = line[6]
+            output = (gene, pvalue)
+            # write to output file
+            differential_out.write("\t".join(output) + "\n")
```

Comment:
I see duplicates for genes in the differential output (this feels like a problem we've had to address before...). I think in other workflows we pick the smallest p-value per-gene to report (e.g. if tests multiple diff transcript models of the same gene). Could we also do something similar here?

![image](https://user-images.githubusercontent.com/49978382/162204708-d3016ed1-5f35-43c5-8581-19f4f89e89d6.png)

Also not sure about whether version numbers for gene IDs being absent here will be a problem or not (e.g. 'ENSMUSG00000036206.1'). I think for other workflows **where we control the input annotation generation** we don't strip the version numbers, and I assume for ground truth data we keep it as is. I don't have a preference in either direction, it's just we have to be consistent with the ground truth data (or adapt how the summary workflow matches gene IDs) such that the gene IDs will match. I understand that this is out of our control with CSI-UTR as we're currently using the pre-provided annotations, but this still needs to be documented

Reply:
Agree, we definitely need to remove the duplicates. For the test data I get over 50'000 rows in `control_vs_srsf3_differetinal_output.tsv` with some genes appearing 5 times or more. 

As for the version numbers: not sure how to handle this right now, but we can always strip the version number in the summary workflow. I agree, a note in the README that the genes are without version number should be there and I hope that suffices for now.